Database-Features/removing-corrupt-data.md

```diff
@@ -0,0 +1,83 @@
+
+Note: Local backups which were written before the event may not be reliable. There is no way to verify if the data blocks from the local backup were corrupted or not. For this reason, the backup to restore from should be stored remotely. 
+
+For information on restoring the database from a backup, see [Restore Database from Backup](https://docs.exasol.com/db/latest/administration/on-premise/backup_restore/restore_database.htm).
+
+### 2. Identify and Delete Corrupt Data
+It's possible to try to identify the objects which have the corrupt data. Depending on the size of the database, this option may be very time-consuming, and in the end, may not find all corrupt data. The following scripts will only help find corrupt data that is stored in a table/column, statistics, or indexes. If data corresponding to other data structures which are not checked in the below scripts are corrupt, then it may not be possible to remove this data. This option gives you the chance to "save" the database and not have to perform a restore, **but does not provide any guarantees**. 
+
+Save the below script into a file and run the below script using [Exaplus CLI](https://docs.exasol.com/db/latest/connect_exasol/sql_clients/exaplus_cli/exaplus_cli.htm). It will generate a list of SQL statements which forces the database to read all columns of every table in the entire database. 
+
+```sql
+set heading off;
+set verbose off;
+set linesize 20000;
+spool /tmp/check_all_columns.sql;
+SELECT '/*"' || COLUMN_SCHEMA || '"."' || COLUMN_TABLE || '"*/ ' || 'SELECT '||GROUP_CONCAT(CASE WHEN column_type='BOOLEAN' THEN 'COUNT("'||column_name||'")' ELSE CASE WHEN column_type LIKE '%CHAR%' THEN 'MIN(LENGTH("'||column_name||'"))' ELSE 'MIN("'||column_name||'")' END END ORDER BY column_ordinal_position)||' FROM "'||column_schema||'"."'||column_table||'";' FROM SYS."$EXA_SYS_COLUMNS_BASE" where column_schema='EXA_STATISTICS' and column_object_type='TABLE' group by column_table,column_schema order by column_table desc;
+SELECT '/*"' || COLUMN_SCHEMA || '"."' || COLUMN_TABLE || '"*/ ' || 'SELECT '||GROUP_CONCAT(CASE WHEN column_type='BOOLEAN' THEN 'COUNT("'||column_name||'")' ELSE CASE WHEN column_type LIKE '%CHAR%' THEN 'MIN(LENGTH("'||column_name||'"))' ELSE 'MIN("'||column_name||'")' END END ORDER BY column_ordinal_position)||' FROM "'||column_schema||'"."'||column_table||'";' FROM SYS.EXA_DBA_COLUMNS where column_object_type='TABLE' group by column_table,column_schema order by column_table desc;
+spool /tmp/check_all_columns.log;
+set heading on;
+set verbose on;
+@/tmp/check_all_columns.sql;
+```
+To execute the file, run `exaplus -u sys -p <passwd> -c <connection string> -f <file name>`. 
```

Comment:
I think here is another logical error.
There are two scripts we run:
1. The generator above.
2. The /tmp/check_all_columns.sql script to check the database. 

I believe the text could make it much clearer what to run and when. The check_all_columns.sql should be in the text / exaplus commands but it is not.

Reply:
Added text to clarify which script